File: Sort/561_Array-Partition-I/561. Array Partition I.cpp
/*******************************************************************************
 * Title:
 *     561. Array Partition I
 *     561. 数组拆分 I
 * Address:
 *     https://leetcode-cn.com/problems/array-partition-i/
 ******************************************************************************/

#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

// 方法一：
class Solution {
public:
    int arrayPairSum(vector<int>& nums) {
        sort( nums.begin(), nums.end() );

        int sum = 0;
        for ( int i = 0; i < (int)nums.size(); i += 2 ) 
            sum += nums[i];
        
        return sum;
    }
};


int main()
{
    

    return 0;
}

File: Sort/561_Array-Partition-I/561. Array Partition I.py
"""
 * Title:
 *     561. Array Partition I
 *     561. 数组拆分 I
 * Address:
 *     https://leetcode-cn.com/problems/array-partition-i/
"""

# 方法一：排序
class Solution:
    def arrayPairSum(self, nums: List[int]) -> int:
        nums.sort()
        
        ans = 0
        for i in range(0, len(nums), 2):
            ans += nums[i]
        
        return ans

# 方法二：排序
class Solution_2:
    def arrayPairSum(self, nums: List[int]) -> int:
        nums.sort()

        return sum(nums[::2])